source/SIL.AppBuilder.Portal.Frontend/src/ui/components/inputs/user-select/with-data.tsx

```diff
@@ -1,117 +1,169 @@
+import {
+  query, defaultSourceOptions, OrganizationResource,
+  UserResource,
+  GroupMembershipResource
+} from '@data';
+
+import { TYPE_NAME as USER } from '@data/models/user';
 import { withCurrentUser } from '@data/containers/with-current-user';
+import { retrieveRelation } from '@data/containers/with-relationship';
+// import { roleInOrganization } from '@data/containers/with-role';
 
+import { PageLoader as Loader } from '@ui/components/loaders';
 
 export interface IProvidedProps {
-  users: Array<ResourceObject<USERS_TYPE, UserAttributes>>;
+  users: UserResource[];
   disableSelection: true;
 }
 
 interface IOwnProps {
-  users: Array<ResourceObject<USERS_TYPE, UserAttributes>>;
-  groupMemberships: Array<ResourceObject<GROUP_MEMBERSHIPS_TYPE, GroupMembershipAttributes>>;
-  currentUsersGroupMemberships: Array<ResourceObject<GROUP_MEMBERSHIPS_TYPE, GroupMembershipAttributes>>;
-  usersFromCache: Array<ResourceObject<USERS_TYPE, UserAttributes>>;
-  currentUser: ResourceObject<USERS_TYPE, UserAttributes>;
+  users: UserResource[];
+  groupMemberships: GroupMembershipResource[];
+  currentUsersGroupMemberships: GroupMembershipResource[];
+  usersFromCache: UserResource[];
+  currentUser: UserResource;
   selected: Id;
   groupId: Id;
   restrictToGroup: boolean;
+  scopeToOrganization?: OrganizationResource;
 }
 
 type IProps =
   & IOwnProps
   & WithDataProps;
 
-export function withData(WrappedComponent) {
-  const mapNetworkToProps = (passedProps) => {
-    return {
-      users: [
-        q => q.findRecords(USER), {
-          label: 'Get Users for User Input Select',
-          sources: {
-            remote: {
-              settings: { ...defaultSourceOptions() },
-              include: ['group-memberships'],
-            }
-          }
-        }
-      ]
-    };
-  };
-
-  const mapRecordsToProps = (passedProps) => {
-    const { type, id } = passedProps.currentUser;
+interface IState {
+  users?: UserResource[];
+}
 
-    return {
-      currentUsersGroupMemberships: q => q.findRelatedRecords({ type, id }, 'groupMemberships'),
-      groupMemberships: q => q.findRecords('groupMembership')
-    };
-  };
+export function withData(WrappedComponent) {
+  class DataWrapper extends React.Component<IProps, IState> {
+    isFilteringUsers = false;
+    state: IState = {};
 
-  class DataWrapper extends React.Component<IProps> {
-    render() {
+    determineAvailableUsers = async () => {
       const {
+        dataStore,
         users,
-        currentUser,
         selected,
         groupId,
-        groupMemberships,
+        scopeToOrganization,
         restrictToGroup,
-        currentUsersGroupMemberships,
-        ...otherProps
       } = this.props;
 
-      if (!users || !groupMemberships || !currentUsersGroupMemberships) {
-        return <Loader />;
+      // remove users who _can't_ be assigned to this project
+      // due to not having organization overlap
+      const filtered = [];
```

Comment:
refactored all of this, because the old code was pretty difficult to reason about